subsys/bluetooth/shell/csis_client.c

```diff
@@ -76,8 +68,27 @@ static void connected_cb(struct bt_conn *conn, uint8_t err)
 		   bt_conn_index(conn));
 }
 
+static void disconnected_cb(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Disconnected: %s (reason 0x%02x)\n", addr, reason);
+
+	for (uint8_t i = 0; i < members_found; i++) {
+		if (set_members[i].conn == conn) {
+			bt_conn_unref(set_members[i].conn);
+			set_members[i].conn = NULL;
+			members_found--;
```

Comment:
I assume you want to loop until all the members have been found, but here you decrease 'members_found', so you may leave the loop too early

Reply:
A `set_member` represents a remote device with a CSIS. There is a 1:1 relation between a `set_member` and an `bt_conn`, so there should only ever be a single `set_member` where `set_members[i].conn == conn`

Reply:
Thanks for the explanation; I see what's happening now